lib/puppet/util/character_encoding.rb

```diff
@@ -77,22 +77,5 @@ def override_encoding_to_utf_8(string)
       Encoding::UTF_8 => "\uFFFD",
       Encoding::UTF_16LE => "\xFD\xFF".force_encoding(Encoding::UTF_16LE),
     }
-
-    # Given a string, return a copy of that string with any invalid byte
-    # sequences in its current encoding replaced with the replacement character
-    # "\uFFFD" (UTF-8) if the string is UTF-8 or UTF-16LE, or "?" otherwise.
-    # @param string a string to remove invalid byte sequences from
-    # @return a copy of string invalid byte sequences replaced by the unicode
-    #   replacement character or "?" character
-    # @note does not modify encoding, but new string will have different bytes
-    #   from original. Only needed for ruby 1.9.3 support.
-    def scrub(string)
-      if string.respond_to?(:scrub)
-        string.scrub
-      else
-        replacement_character = REPLACEMENT_CHAR_MAP[string.encoding] || '?'
```

Comment:
I think you can also remove the `REPLACEMENT_CHAR_MAP` constant since I think this is the only caller.